sample/src/main/java/com/example/AfterpayV3SampleActivity.kt

```diff
@@ -0,0 +1,132 @@
+            /**
+             * Step 3: Respond to intent result and show error if [android.app.Activity.RESULT_CANCELED]
+             * or signPayment with token if [android.app.Activity.RESULT_OK]
+             */
+            val intent = result.data
+            checkNotNull(intent)
+            if (result.resultCode == RESULT_OK) {
+                Afterpay.parseCheckoutSuccessResponseV3(intent)?.let {
+                    /**
+                     * Step 4: Receive single-use card details. Pass these back to your server and on
+                     * to your payment processor:
+                     */
+                    it.cardDetails
+                    it.tokens
+                    it.cardValidUntil
+
+                    showToast(this, "Payment details received. Ready to process payment")
+                }
+            } else if (result.resultCode == RESULT_CANCELED) {
+                Afterpay.parseCheckoutCancellationResponseV3(intent)
+                    ?.let {
+                            (
+                                cancellationStatusV3: CancellationStatusV3,
+                                exception: Exception?,
+                            ),
+                        ->
+                        Log.e(tag, cancellationStatusV3.toString(), exception)
+                    }
+            }
+        }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        bindings = AfterpayV3LayoutBinding.inflate(LayoutInflater.from(this))
+
+        bindings.afterpayButton.apply {
+            setOnClickListener { view ->
+                /**
+                 * Step 2: Once configuration is set, Afterpay SDK will automatically enable button,
+                 * allowing customer to start checkout flow
+                 *
+                 * Create [CheckoutV3Configuration], create an [Intent], and start flow.
+                 *
+                 * Here we use Android's ActivityResult APIs but you can use older [startActivityForResult]
+                 */
+                val intent =
+                    Afterpay.createCheckoutV3Intent(
+                        context = view.context,
+                        consumer = createConsumer(),
+                        orderTotal = createOrderTotal(),
+                        items = createItems(),
+                        buyNow = true,
+                        configuration = createCheckoutV3Configuration(),
+                    )
+                activityResultLauncher.launch(intent)
+            }
+        }
+
+        /**
+         * Step 0: Periodically check for new Merchant configurations. Doesn't need
+         * to happen before *every* transaction. Does need to happen before *first* transaction
+         * Use the tool of your choice to enable async non-main-thread network request (i.e. coroutines, rxjava)
+         *
+         * Using [lifecycleScope] is a quick-and-dirty example. Ideally you would not tie this network
```

Comment:
Are you planning on moving some of this code into a Google's `ViewModel` ? I'm thinking that aligning with Google's official Android docs is probably the easiest to guarantee minimal knowledge ramp-up for potential adopters, and would improve this class readability and stability in regards to lifecycle events.